From f47f18c2f666cea2d20a60afe9255ab93ae243bc Mon Sep 17 00:00:00 2001
From: Marco Capetta <mcapetta@sipwise.com>
Date: Thu, 29 Dec 2022 10:14:20 +0100
Subject: MT#55831 NGCP refreshes the session even if STT is not supported

Unset Sems internal Session Timers in case B leg doesn't support it
(no Timer in Supported header of 200OK message).
(real ticket number: TT#29926)

--- a/core/plug-in/session_timer/SessionTimer.cpp
+++ b/core/plug-in/session_timer/SessionTimer.cpp
@@ -346,7 +346,21 @@ void SessionTimer::updateTimer(AmSession
   if (((reply.code < 200) || (reply.code >= 300)) &&
       (!(accept_501_reply && reply.code == 501)))
     return;
-  
+
+  // verify if B leg supports Session Timers
+  remote_timer_aware =
+      key_in_list(getHeader(reply.hdrs, SIP_HDR_SUPPORTED, SIP_HDR_SUPPORTED_COMPACT),
+                  TIMER_OPTION_TAG);
+
+  if (!remote_timer_aware) {
+    // timer NOT supported by B leg
+    DBG("Session Timer NOT supported by leg B, removing internal session timer intervals");
+    session_timer_conf.setEnableSessionTimer("no");
+    removeTimers(s);
+    return;
+  }
+
+  // timer supported by B leg
   // determine session interval
   string sess_expires_hdr = getHeader(reply.hdrs, SIP_HDR_SESSION_EXPIRES,
 				      SIP_HDR_SESSION_EXPIRES_COMPACT, true);
